### PR TITLE
Introduce elemInstances option

### PR DESCRIPTION
**Pull request by tadatuta**

### CODE REVIEW

**Comment by tadatuta** [comment]
cc @miripiruni @veged

File: docs/ru/3-api.md [hunk 1]
```diff
@@ -138,6 +138,33 @@ var html = templates.apply(bemjson);
 ```
 Подробнее про [cоглашение по именованию](https://ru.bem.info/method/naming-convention/) на bem.info.
 
+### Поддержка JS-экземпляров для элементов (bem-core v4+)
+В библиотеке [bem-core](https://ru.bem.info/libs/bem-core/) начиная с 4 версии появилась поддержка JS-экземпляров для элементов.
+Это требует добавления класса `i-bem` для элементов, имеющих js-параметры.
```
Comment (miripiruni) [inline docs/ru/3-api.md]: JS-параметры

File: test/elemInstances-test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+var fixtures = require('./fixtures')('bemhtml');
+var test = fixtures.test;
+
+describe('elemInstances option', function() {
```
Comment (miripiruni) [inline test/elemInstances-test.js]: Here is something else:

```js
{
  block: 'a',
  js: true,
  mix: { block: 'b', elem: 'e', js: true }
}
``` 
and

```js
{
  block: 'b',
  elem: 'e',
  js: true,
  mix: { block: 'a', js: true }
}
```

File: test/elemInstances-test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+var fixtures = require('./fixtures')('bemhtml');
```
Comment (miripiruni) [inline test/elemInstances-test.js]: Can you move these tests to `bemjson-js-test.js`?

**Comment by miripiruni** [comment]
Looks like we need add tests [to modes](https://github.com/bem/bem-xjst/blob/master/test/modes-js-test.js) for `block('b').elem('e')`…

**Comment by tadatuta** [comment]
> Looks like we need add tests to modes for block('b').elem('e')…

I can but last time I did it for `xhtml` option @veged asked to remove as there's the same logic under the hood (see: https://github.com/bem/bem-xjst/pull/206#discussion-diff-57181009).

**Comment by miripiruni** [comment]
@tadatuta we need to be sure that `js` mode works for `elem`.

File: docs/ru/3-api.md [hunk 4]
```diff
@@ -138,6 +138,33 @@ var html = templates.apply(bemjson);
...
+Это требует добавления класса `i-bem` для элементов, имеющих JS-параметры.
+Добиться этого можно с помощью опции `elemInstances`:
+```js
+var bemxjst = require('bem-xjst');
+var templates = bemxjst.bemhtml.compile(function() {
+    // В этом примере мы не добавляем пользовательских шаблонов.
+    // Для рендеринга HTML будет использовано поведение шаблонизатора по умолчанию.
+    }, {
+        // Включаем поддержку JS-экземпляров для элементов
+        elemInstances: true
```
Comment (veged) [inline docs/ru/3-api.md]: maybe `elemJsInstances`?

**Comment by veged** [comment]
`js` mode already should works for elems (in term of `data-bem` attribute) — this issue only about `i-bem` class

I mean, if tests about `js` mode are really needed semantically better to add them in separate PR

**Comment by veged** [comment]
except `elemJsInstances` vs. `elemInstances` LGTM

**Comment by tadatuta** [comment]
:up:

**Comment by miripiruni** [comment]
LGTM

**Comment by veged** [comment]
@miripiruni can you make new minor version with this?

**Comment by miripiruni** [comment]
@veged  [v6.3.0](https://www.npmjs.com/package/bem-xjst) already in production :)

**Comment by veged** [comment]
:+1:

**Pull request merged**